lib/tlsca/ca.go

```diff
@@ -1353,3 +1354,10 @@ func (ca *CertAuthority) GenerateCertificate(req CertificateRequest) ([]byte, er
 
 	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certBytes}), nil
 }
+
+// shouldPersistJoinAttrs returns true if the join attributes should be persisted
+// into the X509 identity. This provides an emergency "off" handle for this
+// new behavior until we are confident it is working as expected.
+func shouldPersistJoinAttrs() bool {
+	return os.Getenv("TELEPORT_UNSTABLE_DISABLE_JOIN_ATTRS") != "1"
+}
```

Comment:
A bit unfortunate naming. Feels like unstable should be opt-in, not opt-out, but I don't have anything better.

Reply:
Hm yeah - I suppose what I meant by "unstable" here is less that the feature is unstable, and more that the env var is unstable (e.g may be removed in future). This is in line with other env vars we have such as:

- TELEPORT_UNSTABLE_SKIP_VERSION_UPGRADE_CHECK
- TELEPORT_UNSTABLE_DISABLE_UNQUALIFIED_LOOKUPS
- TELEPORT_UNSTABLE_DISABLE_INSTANCE_HB
- TELEPORT_UNSTABLE_DISABLE_VARIABLE_RATE_HEARTBEATS